### DISCUSSION TITLE
Document common Nitro front end development configuration

### DISCUSSION THREAD

**Discussion by mattstein** [craftcms/docs#308] [Ideas]
### Description

Nitro [exposes ports 3000 and 3001](https://craftcms.com/docs/nitro/2.x/node.html#configuring-node-server-ports) for allowing in-container node servers to talk with browsers on the outside. How you’d use this in your setup depends on the details of your setup, so it’d be ideal to have one or more Knowledge Base articles covering the details of common Craft front end development setups.

What _are_ the most common front end development setups? Would love to get a better sense of that to focus these articles.

**Comment by ademers** [comment]
Based on what I've seen in the Craft Discord, many people, myself included,  are using Laravel Mix + Twigpack. Thanks for your consideration!

**Comment by kevinmu17** [comment]
Bare minimum with only webpack + HMR configured together with AssetRev or twigpack.

**Comment by proimage** [comment]
To date, we've been using Craft + Laravel Mix + the [Mix plugin (by mister bk)](https://github.com/mister-bk/craft-plugin-mix). I might have to check out this Twigpack of which y'all speak, but first: getting Nitro working on Windows! 🤢
